test/build/runner.js

```diff
@@ -0,0 +1,84 @@
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs').promises;
+const {readdirSync, readFileSync, statSync, existsSync} = require('fs');
+const execa = require('execa');
+const dircompare = require('dir-compare');
+
+function stripWhitespace(stdout) {
+  return stdout.replace(/((\s+$)|((\\r\\n)|(\\n)))/gm, '');
+}
+function stripRev(code) {
+  return code.replace(/\?rev=\w+/gm, '?rev=XXXXXXXXXX');
+}
+function stripChunkHash(stdout) {
+  return stdout.replace(/([\w\-]+\-)[a-z0-9]{8}(\.js)/g, '$1XXXXXXXX$2');
+}
+
+beforeAll(() => {
+  // Needed so that ora (spinner) doesn't use platform-specific characters
+  process.env = Object.assign(process.env, {CI: '1'});
+});
+
+for (const testName of readdirSync(__dirname)) {
+  if (testName === 'node_modules' || testName.includes('.')) {
+    continue;
+  }
+
+  test(testName, async () => {
+    let cwd = path.join(__dirname, testName);
+
+    // add exception for yarn-workspaces test
+    if (testName === 'yarn-workspaces') {
```

Comment:
This is also a weird exception here: because we test out a Yarn workspace repo, it’s tough here to know which folders to compare. We might need to make other exceptions in the future, but maybe this works for now?

Reply:
Fine for now, I think we have a similar concept in the install test dir